includes/templates/class-amp-post-template.php

```diff
@@ -164,10 +148,41 @@ public function __construct( $post ) {
 	 * @return mixed Value.
 	 */
 	public function get( $property, $default = null ) {
+
 		if ( isset( $this->data[ $property ] ) ) {
 			return $this->data[ $property ];
 		}
 
+		// @todo Some of the fields here were previously constructed and filtered in \AMP_Post_Template::__construct(). This may pose back-compat issues.
+		switch ( $property ) {
+			case 'amp_post_content':
+				/** This filter is documented in wp-includes/post-template.php */
+				return apply_filters( 'the_content', get_the_content( null, false, $this->post ) );
+			case 'post_title':
+				return get_the_title( $this->post );
+			case 'document_title':
+				return wp_get_document_title();
+			case 'canonical_url':
+				return get_permalink( $this->ID );
+			case 'home_url':
+				return home_url( '/' );
+			case 'blog_name':
+				return get_bloginfo( 'name', 'display' );
+			case 'post_author':
+				return get_userdata( $this->post->post_author );
+			case 'post_publish_timestamp':
+				return get_the_date( 'U', $this->ID );
+			case 'post_modified_timestamp':
+				return get_post_modified_time( 'U', false, $this->post );
+			case 'post':
+				return $this->post;
+			case 'post_id':
+				return $this->post->ID;
+			case 'site_icon_url':
+				/** This filter is documented in includes/amp-helper-functions.php */
+				return apply_filters( 'amp_site_icon_url', function_exists( 'get_site_icon_url' ) ? get_site_icon_url( self::SITE_ICON_SIZE ) : '' );
+		}
```

Comment:
This makes me nervous. By making the fields here lazy-loaded when invoking the getter, it bypasses the `amp_post_template_data` filter. Perhaps this should be reverted, and the introduction of regular WordPress template tags be replaced with `AMP_Post_Template::get()` to ensure no breakage of backwards-compatibility. See search for `amp_post_template_data` filter across plugins:  https://wpdirectory.net/search/01DSTQ6DK2GCDKF9QDEN0E8X7R

Nevertheless, going forward sites would be free to use WordPress template tags when writing Reader mode templates.

Reply:
Undone in aed6c0e6550ac9404fa96b838b6ab1ddd541a933